### PR TITLE
Migrate tasks/test_composer.py to PyTest

### LINKED ISSUE #3590: Convert bodhi/tests/server/tasks/test_composer.py to PyTest
[bodhi/tests/server/tasks/test_composer.py](https://github.com/fedora-infra/bodhi/blob/develop/bodhi/tests/server/tasks/test_composer.py) still contains classes that derive from `unittest.TestCase`. We use PyTest, so they could be migrated to not derive from `unittest.TestCase`, or derive from `bodhi.tests.server.base.BasePyTestCase` in case they need to use the test database or WSGI app. 

This also allows the use of the simpler assert constructs instead of the self.assert*() methods. Instead of unittest `setUp()` or `tearDown()` methods, PyTest uses `setup_method(...)` or `teardown_method(...)`, or analog methods to setup/teardown stuff for different scopes.

The following PR is an example of the changes that might need to be done to this file:
#3612

### PR DESCRIPTION
**Pull request by Zlopez**
Removes unittest.TestCase dependency

Fixes #3590

Signed-off-by: Michal Konečný <mkonecny@redhat.com>

### CODE REVIEW

**Review by ryanlerch — APPROVED** [review]

**Comment by ryanlerch** [comment]
the unit tests on this one seem to be failing now :(

**Comment by nphilipp** [comment]
I'll rebase this on the current HEAD to restart the tests.

File: bodhi/tests/server/tasks/test_composer.py [hunk 1]
```diff
@@ -108,10 +108,10 @@ def __init__(self):
             def dont_wrap_me_bro(self):
                 return "I told you not to do this. Now look what happened."
 
-        with self.assertRaises(ValueError) as exc:
+        with pytest.raises(ValueError) as exc:
             TestClass().dont_wrap_me_bro()
 
-        self.assertEqual(str(exc.exception), 'checkpointed functions may not return stuff')
+        assert 'checkpointed functions may not return stuff' in str(exc)
```
Comment (nphilipp) [inline bodhi/tests/server/tasks/test_composer.py]: The `pytest.raises()` context manager is a wrapper around a potentially thrown exception similar to `unittest.TestCase.assertRaises()`, the assertion should look like this:

````
    assert 'checkpointed functions may not return stuff' in exc.value
````

This needs to be fixed in a couple other cases below, too.

File: bodhi/tests/server/tasks/test_composer.py [hunk 2]
```diff
@@ -170,7 +170,6 @@ def setUp(self):
         mock_config = mock.patch.dict(
             'bodhi.server.tasks.composer.config', test_config)
         mock_config.start()
-        self.addCleanup(mock_config.stop)
```
Comment (nphilipp) [inline bodhi/tests/server/tasks/test_composer.py:173]: This should be moved into `teardown_method()`, like this:

````
    def setup_method(self, method):
        ...
        self.mock_config = mock.patch.dict(...)
        self.mock_config.start()
    ...
    def teardown_method(self, method):
        super().teardown_method(method)
        self.mock_config.stop()
        ...
````

File: bodhi/tests/server/tasks/test_composer.py [hunk 3]
```diff
@@ -2344,14 +2324,14 @@ def test_request_not_stable(self, Popen):
                        'testcontainer2:1.0.1-1.fc28container'):
             for dtag in [source.split(':')[1], source.split(':')[1].split('-')[0], 'testing']:
                 mock_call = mock.call(
-                    [config['skopeo.cmd'], 'copy',
+                    [config['skopeo.cmd'], 'copy', '--dest-tls-verify=false',
```
Comment (nphilipp) [inline bodhi/tests/server/tasks/test_composer.py]: This looks like it's mistakenly copied from `test_skopeo_extra_copy_flags()`. It breaks this and a couple of other tests for me, at any rate. :wink:

**Review by nphilipp — CHANGES_REQUESTED** [review]
Here are the changes I made to fix the unit tests:

[PR3590-changes.patch.txt](https://github.com/fedora-infra/bodhi/files/3824115/PR3590-changes.patch.txt)

File: bodhi/tests/server/tasks/test_composer.py (hunk 2 above)
Comment (Zlopez) [reply bodhi/tests/server/tasks/test_composer.py:173]: Thanks, I couldn't find out how to call this.

File: bodhi/tests/server/tasks/test_composer.py (hunk 1 above)
Comment (Zlopez) [reply bodhi/tests/server/tasks/test_composer.py]: I noticed this today, when checking other PRs.

File: bodhi/tests/server/tasks/test_composer.py (hunk 3 above)
Comment (Zlopez) [reply bodhi/tests/server/tasks/test_composer.py]: I will remove it than.

File: bodhi/tests/server/tasks/test_composer.py [hunk 4]
```diff
@@ -2656,10 +2642,8 @@ def test_primary_arches_undefined(self):
 
         url = t._get_master_repomd_url('aarch64')
 
-        self.assertEqual(
-            url,
+        assert url ==\
```
Comment (nphilipp) [inline bodhi/tests/server/tasks/test_composer.py]: Please add a space between `==` and the backslash (see PEP8: "Always surround these binary operators with a single space on either side..."), here and in similar code below. Sorry for not spotting this earlier.

**Review by nphilipp — CHANGES_REQUESTED** [review]

File: bodhi/tests/server/tasks/test_composer.py (hunk 4 above)
Comment (Zlopez) [reply bodhi/tests/server/tasks/test_composer.py]: Why flake8 didn't catch this?

Comment (Zlopez) [reply bodhi/tests/server/tasks/test_composer.py]: This should be fixed now, I found a few more in the code so I fixed them too.

Comment (nphilipp) [reply bodhi/tests/server/tasks/test_composer.py]: I think flake8 didn't catch this because of the backslash line-continuation.

**Review by nphilipp — APPROVED** [review]

**Pull request merged**